src/DataStructures/Tensor/Expressions/Evaluate.hpp

```diff
@@ -9,9 +9,34 @@
 #include "DataStructures/Tensor/Expressions/LhsTensorSymmAndIndices.hpp"
 #include "DataStructures/Tensor/Expressions/TensorExpression.hpp"
 #include "DataStructures/Tensor/Tensor.hpp"
+#include "Utilities/Gsl.hpp"
 #include "Utilities/Requires.hpp"
 
 namespace TensorExpressions {
+
+namespace detail {
+template <size_t NumIndices>
+constexpr bool contains_indices_to_contract(
+    const std::array<size_t, NumIndices>& tensorindices) noexcept {
+  // Special case handling needed here because constexpr step limit is reached
+  // for NumIndices == 0 in nested loops in else block
+  if constexpr (NumIndices < 2) {
+    return false;
+  } else {
+    for (size_t i = 0; i < NumIndices - 1; i++) {
+      for (size_t j = i + 1; j < NumIndices; j++) {
```

Comment:
I'm not quite sure why when `NumIndices == 0`, the constexpr step limit is reached on this line (when the `if` block is removed). I wanted to implement this using `std::array`s instead of `tmpl` functions just to reduce the number of instantiations of `contains_indices_to_contract`, but I'm happy to implement this function another way, if preferred!

Reply:
There is either a logical error I've made that I can't see, or the compiler gets confused with how to handle the inner loop when the outer loop isn't entered in the first place